// twitter-v1-oauth/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.encode = void 0;
const tslib_1 = require("tslib");
const querystring_1 = tslib_1.__importDefault(require("querystring"));
const helpers_1 = require("./helpers");
const signature_1 = tslib_1.__importDefault(require("./modules/signature"));
function encode(data) {
    return querystring_1.default.stringify(data, "&", "=", {
        encodeURIComponent: helpers_1.percentEncode,
    });
}
exports.encode = encode;
function oAuth1a(requestOptions, oAuthOptions) {
    const oAuthParams = {
        oauth_consumer_key: oAuthOptions.api_key,
        oauth_nonce: (0, helpers_1.randomString)(32),
        oauth_signature: "",
        oauth_signature_method: "HMAC-SHA1",
        oauth_timestamp: (0, helpers_1.timestamp)(),
        oauth_token: oAuthOptions.access_token,
        oauth_version: "1.0",
    };
    oAuthParams.oauth_signature = (0, signature_1.default)({
        ...requestOptions,
        oAuthOptions: { ...oAuthOptions, ...oAuthParams },
    });
    return `OAuth ${Object.entries(oAuthParams)
        .map(([key, value]) => `${(0, helpers_1.percentEncode)(key)}="${(0, helpers_1.percentEncode)(String(value))}"`)
        .join(", ")}`;
}
exports.default = oAuth1a;
